src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -397,23 +397,21 @@ public void ExecuteSuppressionAction(DiagnosticSuppressor suppressor, ImmutableA
         {
             Debug.Assert(compilationEvent is CompilationStartedEvent || compilationEvent is CompilationCompletedEvent);
 
-            AnalyzerStateData? analyzerState = null;
-
-            try
+            if (TryStartProcessingEvent(compilationEvent, analyzer, analysisScope, analysisState, out var analyzerState))
             {
-                if (TryStartProcessingEvent(compilationEvent, analyzer, analysisScope, analysisState, out analyzerState))
+                try
                 {
                     ExecuteCompilationActionsCore(compilationActions, analyzer, analyzerState);
-                    analysisState?.MarkEventComplete(compilationEvent, analyzer);
-                    return true;
+                }
+                finally
+                {
+                    analyzerState?.ResetToReadyState();
                 }
 
-                return IsEventComplete(compilationEvent, analyzer, analysisState);
-            }
-            finally
-            {
-                analyzerState?.ResetToReadyState();
+                return TryMarkEventComplete(compilationEvent, analyzer, analysisState);
```

Comment:
>return TryMarkEventComplete(compilationEvent, analyzer, analysisState); [](http://example.com/codeflow?start=16&length=71)

Is the change to have `MarkEventComplete()` return a `bool` and return that here, just an optimization over calling `return IsEventComplete();` in all cases? (I think that is a reasonable change, but just confirming that I understand this.)

Reply:
Yes, that is correct. We could split this into two calls, but that is unnecessary.